azkaban-common/src/main/java/azkaban/jobExecutor/ProcessJob.java

```diff
@@ -243,7 +243,14 @@ public void run() throws Exception {
       // Set parent directory permissions to <uid>:azkaban so user can write in their execution directory
       // if the directory is not permissioned correctly already (should happen once per execution)
       if (!canWriteInCurrentWorkingDirectory(effectiveUser)) {
-        assignUserDirOwnership(effectiveUser);
+        info("Changing current working directory ownership");
+        assignUserFileOwnership(effectiveUser, getWorkingDirectory());
```

Comment:
I'm seeing the method `getWorkingDirectory` requires a public modifier, but `protected` should be good enough.

Reply:
`getWorkingDirectory` is legacy code (last updated in 2012) and I'm concerned there might be plugins or something that use it. So I'd rather not update it in this PR. Does that make sense?

Reply:
I see!